app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt

```diff
@@ -49,16 +59,55 @@ class AuthenticationViewModel @Inject constructor(
         ApiModule.providesBearerInterceptor().bearerToken = response.data.token
         showCoreActivity.value = true
     }
+
+    override suspend fun register() {
+        if (passwordState.value.text != reEnteredPasswordState.value.text) {
+            toastMessage.value = resource.getString(R.string.match_password)
+            return
+        }
+
+        val registerBody = UserBody(
+            idState.value.text,
+            nameState.value.text,
+            passwordState.value.text,
+        )
+        val response = userRepository.signUp(registerBody)
+
+        if (response.data == null) {
+            toastMessage.value = response.message ?: resource.getString(R.string.unknown_error)
+            return
+        }
+
+        switchToLogin()
+    }
+
+    private fun switchToLogin() {
+        resetValues()
+        loginState.value = true
+        toastMessage.value = resource.getString(R.string.created_account)
+    }
+
+    private fun resetValues() {
+        nameState.value = TextFieldValue()
+        passwordState.value = TextFieldValue()
+        reEnteredPasswordState.value = TextFieldValue()
+    }
 }
 
 class AuthenticationViewModelPreview : IAuthenticationViewModel {
-    override var nameState = mutableStateOf(TextFieldValue("Peter"))
-    override var idState = mutableStateOf(TextFieldValue("4242"))
-    override var passwordState = mutableStateOf(TextFieldValue("1234"))
-    override var reEnteredPasswordState = mutableStateOf(TextFieldValue("1234"))
+    override val nameState = mutableStateOf(TextFieldValue("Peter"))
+    override val idState = mutableStateOf(TextFieldValue("4242"))
+    override val passwordState = mutableStateOf(TextFieldValue("1234"))
+    override val reEnteredPasswordState = mutableStateOf(TextFieldValue("1234"))
     override val showCoreActivity = MutableLiveData<Boolean>()
+    override val toastMessage = MutableLiveData<String>()
+    override val loginState = mutableStateOf(true)
 
     override suspend fun login() {
         TODO("Not yet implemented")
     }
+
+    override suspend fun register() {
+        TODO("Not yet implemented")
+    }
```

Comment:
Shouldn`t those functions be implemented or at least deleted?

Reply:
They can't be deleted, because the Preview need to implement the Interface.
They are not used currently and shouldn't cause any trouble, since the Preview is only used for previewing the Composable.
I didn't delete them, because we need this later for testing (the preview could act as a mock).